router/batchrouter/handle_lifecycle.go

```diff
@@ -126,18 +128,22 @@ func (brt *Handle) Setup(
 	config.RegisterDurationConfigVariable(600, &diagnosisTickerTime, false, time.Second, []string{"Diagnostics.batchRouterTimePeriod", "Diagnostics.batchRouterTimePeriodInS"}...)
 	brt.diagnosisTicker = time.NewTicker(diagnosisTickerTime)
 	brt.uploadedRawDataJobsCache = make(map[string]map[string]bool)
-	brt.asyncDestinationStruct = make(map[string]*asyncdestinationmanager.AsyncDestinationStruct)
 
 	asyncStatTags := map[string]string{
 		"module":   "batch_router",
-		"destType": destType,
+		"destType": destination.DestinationDefinition.Name,
 	}
 	brt.asyncPollTimeStat = stats.Default.NewTaggedStat("async_poll_time", stats.TimerType, asyncStatTags)
 	brt.asyncFailedJobsTimeStat = stats.Default.NewTaggedStat("async_failed_job_poll_time", stats.TimerType, asyncStatTags)
 	brt.asyncSuccessfulJobCount = stats.Default.NewTaggedStat("async_successful_job_count", stats.CountType, asyncStatTags)
 	brt.asyncFailedJobCount = stats.Default.NewTaggedStat("async_failed_job_count", stats.CountType, asyncStatTags)
 	brt.asyncAbortedJobCount = stats.Default.NewTaggedStat("async_aborted_job_count", stats.CountType, asyncStatTags)
 
+	brt.asyncDestinationStruct = make(map[string]*common.AsyncDestinationStruct)
+	if slices.Contains(asyncDestinations, destination.DestinationDefinition.Name) {
+		brt.initAsyncDestinationStruct(destination)
+	}
```

Comment:
Why do we need this when we already do the same via `backendConfigSubscriber` 
```suggestion
```

Reply:
If this function is not required we can just pass `destType` instead of the whole `destination` struct to the Setup function

Reply:
addressing [here](https://github.com/rudderlabs/rudder-server/pull/3623/commits/8ef6b7e62016a365197c3529d5b27d8d43e5d071#diff-2fa84a69a3bfa3b97f23fd9d560c84f2e284dc8c0bdb4222dcc18937e086f687)

Reply:
addressed https://github.com/rudderlabs/rudder-server/pull/3623